src/MoBi.Core/Commands/AddModuleCommand.cs

```diff
@@ -0,0 +1,60 @@
+﻿using MoBi.Assets;
+using MoBi.Core.Domain.Extensions;
+using MoBi.Core.Domain.Model;
+using MoBi.Core.Events;
+using MoBi.Core.Helper;
+using OSPSuite.Core.Commands.Core;
+using OSPSuite.Core.Domain;
+using OSPSuite.Utility.Extensions;
+
+namespace MoBi.Core.Commands
+{
+   public class AddModuleCommand : MoBiReversibleCommand, ISilentCommand
+   {
+      protected Module _module;
+      public string ModuleId { get; private set; }
+      public bool Silent { get; set; }
+
+      public AddModuleCommand(Module module)
+      {
+         ObjectType = new ObjectTypeResolver().TypeFor<Module>();
+         CommandType = AppConstants.Commands.AddCommand;
+         _module = module;
+         ModuleId = module.Id;
+         Description = AppConstants.Commands.AddToProjectDescription(ObjectType, module.Name);
+         Silent = false;
+      }
+
+      protected override void ExecuteWith(IMoBiContext context)
+      {
+         var project = context.CurrentProject;
+         context.Register(_module);
+         _module.AllBuildingBlocks().Each(context.Register);
```

Comment:
I think registering a module should register the building block (visitor implementation). Thoughts?